### DISCUSSION TITLE
bias correction

### DISCUSSION THREAD

**Discussion by vuillaut** [cta-observatory/pyirf#184] [Ideas]
Hi

Is bias correction something wanted in pyirf?

If so, where should it be implemented?
Directly as an option in the resolution functions?
Or as separate functions?

**Comment by maxnoe** [comment]
@vuillaut Yes, I think this is the right place here. But what exactly do you have in mind?

What you'd need is some kind of transformation function `f(reco_energy) -> reco_energy'` that is fit on a test data set. I think MARS uses splines? @moralejo

**Reply by vuillaut** [reply]
My first though is a very simply correction of the bias to centre distributions to 0.
Here is an example of what I had implemented in ctaplot:
https://github.com/cta-observatory/ctaplot/blob/dcb2e332e0ad8138119d543ac81b579b2b5d5ce7/ctaplot/ana/ana.py#L739

**Reply by maxnoe** [reply]
I don't think this will work. The bias correction should definitely be energy dependent for the energy bias correction.

**Comment by HealthyPear** [comment]
Thanks @vuillaut for having activated Discussions - I think now pyirf is already in an advanced status to make this useful

@maxnoe if you didn't know, next time you can use the option in the bottom right menu in the sidebar to create an issue from a discussion. We could migrate to the issue you created or make a new one from this discussion, same for me.

-----------

I think we could generalise this option in a similar way to what we did for resolution:
- a module or a section of it for bias correction methods (or applied to specific cases, like energy)
- the resolution function calls a keyword argument which sets which function to use

Regarding specifically what CTAMARS does see [here](https://cta.cloud.xwiki.com/xwiki/wiki/aswg/view/Main/Benchmarks%20and%20Reference%20Analysis/Step%20by%20step%20reference%20MARS%20analysis/#HAfinalcorrectiontoErec).
This happens after cuts optimisation, so in that case it fits with pyirf's scope.

**Reply by maxnoe** [reply]
I actually used the "REference in new issue feature" to create #185, let's continue there.

**Reply by HealthyPear** [reply]
I think if you use the create issue option the entire chat migrates and not only the reference
